tests/playwright/src/specs/kubernetes-context-smoke.spec.ts

```diff
@@ -0,0 +1,125 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import fs from 'node:fs';
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+import { KubeContextPage } from '../model/pages/kubernetes-context-page';
+import { PreferencesPage } from '../model/pages/preferences-page';
+import { expect as playExpect, test } from '../utility/fixtures';
+
+const testContexts = ['context-1', 'context-2', 'context-3'];
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+test.beforeAll(async ({ runner, welcomePage }) => {
+  runner.setVideoAndTraceName('kube-context-e2e');
+
+  // copy testing kubeconfig file to the expected location
+  const kubeConfigPathSrc = path.resolve(__dirname, '..', '..', 'resources', 'test-kube-config');
+  const kubeConfigPathDst = path.resolve(__dirname, '..', '..', 'resources', 'kube-config');
+  fs.copyFileSync(kubeConfigPathSrc, kubeConfigPathDst);
+
+  await welcomePage.handleWelcomePage(true);
+});
+
+test.afterAll(async ({ runner }) => {
+  test.setTimeout(120000);
+  await runner.close();
+});
+
+test.describe.serial('Verification of kube context management @smoke', () => {
+  test('Load custom kubeconfig in Preferences', async ({ navigationBar }) => {
+    // open preferences page
+    const settingsBar = await navigationBar.openSettings();
+    await settingsBar.expandPreferencesTab();
+    const preferencesPage = await settingsBar.openTabPage(PreferencesPage);
+    await playExpect(preferencesPage.heading).toBeVisible();
+
+    const kubeConfigPathDst = path.resolve(__dirname, '..', '..', 'resources', 'kube-config');
+    await preferencesPage.selectKubeFile(kubeConfigPathDst);
+    await playExpect(preferencesPage.heading).toBeVisible();
+  });
+
+  test('Can load kube contexts in Kubernetes page', async ({ navigationBar }) => {
+    const settingsBar = await navigationBar.openSettings();
+    const kubePage = await settingsBar.openTabPage(KubeContextPage);
+    await playExpect(kubePage.heading).toBeVisible();
+
+    await playExpect.poll(async () => await kubePage.pageIsEmpty(), { timeout: 10000 }).toBeFalsy();
+    for (const context of testContexts) {
+      const row = await kubePage.getContextRowByName(context);
+      if (row === undefined) {
+        throw Error(`Context: '${context}' not found`);
+      }
```

Comment:
`row` here can't ever be undefined, since the only way it can be undefined is if the page is empty, however you are validating that the page is not empty on line 66 as such `getContextRowByName(context)` will always return a `Locator` object here which will satisfy the contition, note that that does not mean that the context is actually available, only that the Locator is.

```suggestion
      const row = await kubePage.getContextRowByName(context);
      await playExpect(row).toBeVisible();
```

Reply:
The `getContextRowByName(context)` was set to return either Locator or Undefined.  Changed to match as it makes sense to only return Locator.